cli/src/action/command.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright 2018-2021 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+use clap::ArgMatches;
+use protobuf::{Message, RepeatedField};
+use reqwest::{blocking::Client, header};
+use serde::Deserialize;
+use transact::protocol::batch::BatchPair;
+use transact::protocol::sabre::ExecuteContractActionBuilder;
+use transact::protos::FromProto;
+use transact::protos::{
+    batch::{Batch, BatchHeader},
+    command::{BytesEntry, Command, CommandPayload, Command_CommandType, GetState, SetState},
+    IntoBytes, IntoProto,
+};
+
+use crate::error::CliError;
+
+use super::{create_cylinder_jwt_auth_signer_key, Action};
+
+pub struct CommandSetStateAction;
+
+impl Action for CommandSetStateAction {
+    fn run<'a>(&mut self, arg_matches: Option<&ArgMatches<'a>>) -> Result<(), CliError> {
+        let args = arg_matches.ok_or(CliError::RequiresArgs)?;
+
+        let key_path = args
+            .value_of("key")
+            .ok_or_else(|| CliError::ActionError("'key' is required".into()))?;
+        let (auth, signer) = create_cylinder_jwt_auth_signer_key(key_path)?;
+
+        let target = args
+            .value_of("target")
+            .ok_or_else(|| CliError::ActionError("'target' is required".into()))?;
+
+        let bytes_entries = args
+            .values_of("state-entry")
+            .ok_or_else(|| CliError::ActionError("'state-entry' is required".into()))?;
+
+        let mut state_writes = Vec::new();
+        let mut addresses = Vec::new();
+
+        for bytes_entry in bytes_entries {
+            let (address, value) = parse_bytes_entry(bytes_entry)?;
+
+            let mut entry = BytesEntry::new();
+            entry.set_key(address.clone());
+            entry.set_value(value);
+
+            state_writes.push(entry);
+
+            addresses.push(address);
+        }
+
+        let mut set_state = SetState::new();
+        set_state.set_state_writes(RepeatedField::from_vec(state_writes));
+
+        let mut command = Command::new();
```

Comment:
Non-blocking comment: This probably should have a builder/into_proto combination.